sui_programmability/examples/games_with_chance/sources/satoshi_flip.move

```diff
@@ -0,0 +1,304 @@
+        let house_data = option::borrow(&game.house_data);
+        house_data.max_bet
+    }
+
+    /// get the minimum stake the house is willing to accept
+    public fun min_bet(game: &Game): u64 {
+        let house_data = option::borrow(&game.house_data);
+        house_data.min_bet
+    }
+    
+    /// On an ended game (that has a non null outcome value) get if player won
+    public fun is_player_winner(game: &mut Game): bool {
+        assert!(option::is_some<Outcome>(&game.outcome), EGameNotEnded);
+        let game_outcome = option::borrow(&game.outcome);
+        game_outcome.player_won
+    }
+
+    /// On an ended game (that has a non null outcome value) get what secret did the house pick
+    public fun secret(game: &mut Game): vector<u8> {
+        assert!(option::is_some<Outcome>(&game.outcome), EGameNotEnded);
+        let game_outcome = option::borrow(&game.outcome);
+        game_outcome.secret
+    }
+
+    /// On an ended game (that has a non null outcome value) get what did the player guess
+    public fun guess(game: &mut Game): u8 {
+        assert!(option::is_some<Outcome>(&game.outcome), EGameNotEnded);
+        let game_outcome = option::borrow(&game.outcome);
+        game_outcome.guess
+    }
+
+    /// Called by the player for a game with null BetData
+    public entry fun bet(game: &mut Game, guess: u8, stake_coin: Coin<SUI>, stake_amount: u64, ctx: &mut TxContext) {
+        assert!(option::is_none(&game.bet_data), EAlreadyAcceptedBet);
+        let house_data = option::borrow(&game.house_data);
+        assert!(stake_amount <= house_data.max_bet, EStakeTooHigh);
+        assert!(stake_amount > house_data.min_bet, EStakeTooLow);
+        assert!(guess == 0 || guess == 1, EGuessNot1Or0);
+        assert!(coin::value(&stake_coin) >= stake_amount, EPlayerCoinNotEnoughBalance);
+        game.epoch = tx_context::epoch(ctx);
+        // Get a balance with the stake_amount value
+        let stake = coin::into_balance(stake_coin);
+        if (balance::value(&stake) > stake_amount) {
+            let total_balance = balance::value(&stake);
+            let to_return = balance::split(&mut stake, total_balance - stake_amount);
+            // return the rest
+            transfer::transfer(coin::from_balance(to_return, ctx), tx_context::sender(ctx));
+        };
+        let bet_data = BetData {
+            stake: stake,
+            guess
+        };
+        option::fill(&mut game.bet_data, bet_data);
+        option::fill(&mut game.player, tx_context::sender(ctx));
+    }
+
+    /// Called by the house either after `fun bet` has been called by some player to end the game,
+    /// either before bet was called to "cancel" the game.
+    ///
+    /// If the house has "forgotten" the secret, then it can use a random secret and it will
+    /// cancel the game, as long as `fun bet` has not been called.
+    public entry fun end_game(game: &mut Game, secret: vector<u8>, ctx: &mut TxContext) {
```

Comment:
consider receiving it by value (ownership) and destructing it
also in cancel_game

Reply:
I can do that with a shared object? Can I use object::delete(`game.id`) then?

Reply:
Right, I see your question on that (I wasn't aware of that limitation...)